### PR TITLE
Feat sdk tests with multi asset

### PR DESCRIPTION
**Pull request by asimpk**

### CODE REVIEW

File: packages/lib-ethers/test/Kumo.test.ts [hunk 1]
```diff
@@ -45,1486 +36,351 @@ const provider = ethers.provider;
...
-        }
-      };
-
-      const hintHelpers = chai.spy.interface({
-        getApproxHint: () => Promise.resolve(fakeHints.shift())
-      });
 
-      const sortedTroves = chai.spy.interface({
-        findInsertPosition: () => Promise.resolve(["fake insert position"])
+  describe("shouldGetPrice", () => {
```
Comment (udosson) [inline packages/lib-ethers/test/Kumo.test.ts:78]: @asimpk - Could you please explain to me why the following tests are part of this test file and why you haven't separated these? Just curious

**Review by udosson — CHANGES_REQUESTED** [review]
@asimpk - The PR already looks very good. Just 1 change and 1 question.

Let's get this merged and then add the system tests with another PR.
Thanks

File: packages/lib-ethers/test/Kumo.test.ts (hunk 1 above)
Comment (asimpk) [reply packages/lib-ethers/test/Kumo.test.ts:78]: > @asimpk - Could you please explain to me why the following tests are part of this test file and why you haven't separated these? Just curious

I seperated only those tests which require `Before` block for the redeployment to get a fresh/clean state for those tests, here these tests don't depend on the clean state that's why I keep these in same file

Comment (udosson) [reply packages/lib-ethers/test/Kumo.test.ts:78]: Alright. But wouldn't it make sense to move the tests to the test files according to their names?

Comment (asimpk) [reply packages/lib-ethers/test/Kumo.test.ts:78]: > Alright. But wouldn't it make sense to move the tests to the test files according to their names?

Yes, we can move them, what do you recommend?

**Review by arthurka-o — COMMENTED** [review]
Can't we extract those identical pieces of code from the every test?
```
 before(async function () {
                if (network.name !== "hardhat") {
                ...
 ```

**Comment by asimpk** [comment]
> if (network.name !== "hardhat"
hi @arthurka-o , yes we can but I didn't test it by extracting this one-liner condition where the context is involved, `before`, `beforeEach` are the hooks provided by mocha, and in the condition is `this.skip`, `this`  refers to Mocha.Context , so I was not sure if it will behave as I expected or not, I left it just as it was before the separation and extracted a few functions in testUtils

**Review by arthurka-o — APPROVED** [review]

**Pull request merged**